core/schema.rs

```diff
@@ -311,6 +312,50 @@ impl Schema {
     }
 }
 
+impl Clone for Schema {
+    fn clone(&self) -> Self {
```

Comment:
instead of implementing `Clone` could we make this a method `clone_deep()` on Schema indicating it copies everything?

Reply:
actually I don't mind the clone trait implementation but a comment would be nice